From: Tomas Henzl <thenzl@redhat.com>
Date: Fri, 6 Nov 2009 15:23:49 +0200
Subject: [scsi] arcmsr: add missing parameter
Message-id: 4AF42365.3000509@redhat.com
O-Subject: [RHEL5.5 PATCH] bz521203 arcmsr: fix - add the missing parameter
Bugzilla: 521203
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Anton Arapov <anton@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

This is for bz#521203

This patch adds a missing 'struct device_attribute *attr'.
to the function definition. The code around this is
not in upstream.
I've verified that with this patch it is possible
to change the value again.
And thanks to Prarit for spotting this.

Tomas

diff --git a/drivers/scsi/arcmsr/arcmsr_hba.c b/drivers/scsi/arcmsr/arcmsr_hba.c
index 1dd9306..d05276f 100644
--- a/drivers/scsi/arcmsr/arcmsr_hba.c
+++ b/drivers/scsi/arcmsr/arcmsr_hba.c
@@ -96,7 +96,9 @@ static void arcmsr_flush_hba_cache(struct AdapterControlBlock *acb);
 static void arcmsr_flush_hbb_cache(struct AdapterControlBlock *acb);
 static const char *arcmsr_info(struct Scsi_Host *);
 static irqreturn_t arcmsr_interrupt(struct AdapterControlBlock *acb);
-static ssize_t arcmsr_adjust_disk_queue_depth(struct device *dev, const char *buf, size_t count)
+static ssize_t arcmsr_adjust_disk_queue_depth(struct device *dev,
+					      struct device_attribute *attr,
+					      const char *buf, size_t count)
 {
 	int queue_depth;
 	struct scsi_device *sdev = to_scsi_device(dev);
